File: domain/src/main/kotlin/com/sidequest/domain/model/Enums.kt
package com.sidequest.domain.model

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

/**
 * Lifecycle status of an [ActionItem]. New items start as [NOT_STARTED].
 */
@Serializable
enum class ActionStatus {
    @SerialName("not_started")
    NOT_STARTED,

    @SerialName("in_progress")
    IN_PROGRESS,

    @SerialName("completed")
    COMPLETED,
}

/**
 * Classification of the content a user shared into the app.
 */
@Serializable
enum class ContentType {
    @SerialName("link")
    LINK,

    @SerialName("text")
    TEXT,

    @SerialName("image")
    IMAGE,

    @SerialName("video_ref")
    VIDEO_REF,
}

File: ios/SideQuestKit/Sources/SideQuestKit/Models/Enums.swift
import Foundation

/// Lifecycle status of an `ActionItem`. New items start as `notStarted`.
///
/// The raw values are the contract's snake_case strings (`backend/api/openapi.yaml`
/// → `ActionStatus`: `not_started | in_progress | completed`) and match the
/// Android client's `@SerialName` values, so the status round-trips identically
/// across platforms.
public enum ActionStatus: String, Codable, CaseIterable {
    case notStarted = "not_started"
    case inProgress = "in_progress"
    case completed = "completed"
}

/// Classification of the content a user shared into the app.
///
/// The raw values are the contract's snake_case strings (`backend/api/openapi.yaml`
/// → `ContentType`: `link | text | image | video_ref`) and match the Android
/// client's `@SerialName` values.
public enum ContentType: String, Codable, CaseIterable {
    case link
    case text
    case image
    case videoRef = "video_ref"
}
